src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/Tokenization.cs

```diff
@@ -342,5 +381,58 @@ private static TokenType MapNodeKindToTokenType(NodeKind kind)
                     return TokenType.Unknown;
             }
         }
+
+        /// <summary>
+        /// Adds token to the given list of tokens if it is feasible (token should not be skipped).
+        /// </summary>
+        /// <param name="tokens">List of tokens to which given token might be added.</param>
+        /// <param name="token">Token to be added if feasible.</param>
+        /// <param name="tokenTypesToSkip">Optional: Token types to be skipped.</param>
+        /// <param name="expression">Optional: Expression.</param>
+        private static void AddTokenIfPossible(this ICollection<ITokenTextSpan> tokens, ITokenTextSpan token, IReadOnlyCollection<TokenType> tokenTypesToSkip = null, string expression = null)
```

Comment:
>void [](http://example.com/codeflow?start=23&length=4)

can we return bool?

Reply:
It is not necessary TBH. It won't be used. This is another Add function on the collection of tokens and Adds don't return bool. don't wanna do something that is not needed. Let me know if you think otherwise